ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -407,6 +401,15 @@ public void rollback() throws org.omg.CORBA.SystemException,
                                     destroyState = false;
                                 }
                                 break;
+                            case XAException.XAER_NOTA:
+                                if (_phaseTwoStarted) {
+                                    // rolled back previously and recovery completed
+                                    destroyState();
+                                    break;
+                                }
+                                // DO NOT add any case handling below here which would break out of the switch and prevent
+                                // XAException.XAER_NOTA from throwing an org.omg.CosTransactions.HeuristicHazard or 
+                                // UNKNOWN exception
```

Comment:
I still think it is confusing to fall through even with the code warning. The only benefit of falling through here is to save three lines of code in the XAException.XAER_NOTA arm.

Reply:
OK, thank you. I'll go ahead and merge this change as what I am doing here is reverting this part you are commenting on back to a point that existed before my original change. I think the code you are referring to was added as this part of a pull request from 2017: https://github.com/jbosstm/narayana/pull/1229/files#diff-c0091e065ebb4dea411f68b3ea81f32d4b0ebe2b1349c2876073d576fdc505b8R422-R427